File: src/Link/Link.ExpertManagement.Domain.Model/Entities/Expert.cs
using Link.Common.Domain.Framework.Models;
using Link.ExpertManagement.Domain.Model.Enums;
using MongoDB.Bson.Serialization.Attributes;

namespace Link.ExpertManagement.Domain.Model.Entities
{
    public sealed class Expert : AggregateRoot<ExpertId>
    {
        public Expert(
            ExpertId id,
            string firstName, 
            string lastName,
            ExpertProfile expertProfile, 
            ExpertStatus status,
            ExpertType type, 
            ExpertContactInfo contactInfo)
        {
            Id = id;
            FirstName = firstName;
            LastName = lastName;
            ExpertProfile = expertProfile;
            Status = status;
            Type = type;
            ContactInfo = contactInfo;
        }

        [BsonElement("firstName")]
        public string FirstName { get; }

        [BsonElement("lastName")]
        public string LastName { get; }

        [BsonElement("expertProfile")]
        public ExpertProfile ExpertProfile { get; }

        [BsonElement("status")]
        public ExpertStatus Status { get; }

        [BsonElement("type")]
        public ExpertType Type { get; }

        public ExpertContactInfo ContactInfo { get; }

        public string FullName => $"{FirstName} {LastName}";
    }
}

File: src/Link/Link.Frontend/Link-Frontend/src/app/models/expert.ts
import { ExpertDto } from '../interfaces/expert-dto';
import { ExpertType } from './expert-type';
import { ExpertStatus } from './expert-status';
export class Expert {
    constructor(
        public id: string,
        public firstName: string,
        public lastName: string,
        public expertType: string,
        public expertStatus: string
    ) { }

    public static Create(dto: ExpertDto): Expert {
        return new Expert(
            dto.id,
            dto.firstName,
            dto.lastName,
            dto.type,
            dto.status);
    }
}
